File: 0145-binary-tree-postorder-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def postArray(self, node, arr):
        if not node:
            return
        if not node.left and not node.right:
            arr.append(node.val)
            return
        else:
            self.postArray(node.left, arr)
            self.postArray(node.right, arr)
            arr.append(node.val)
            return
                
    def postorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        result = list()
        self.postArray(root, result)
        return result
        

File: 0145-binary-tree-postorder-traversal/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void postorder(TreeNode* root, vector<int>& nodeOrder) {
        if (root == nullptr) return;

        if (root->left != nullptr) {
            postorder(root->left, nodeOrder);
        }

        if (root->right != nullptr) {
            postorder(root->right, nodeOrder);
        }

        nodeOrder.push_back(root->val);

    }

    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> ans;
        postorder(root, ans);
        return ans;
    }
};
